skfda/preprocessing/dim_reduction/_fpls.py

```diff
@@ -0,0 +1,829 @@
+                basis_iterable=(
+                    _GridBasis(grid_points=self.data.grid_points),
+                ),
+                regularization_parameter=1,
+                regularization=regularization,
+            )
+        if regularization_matrix is None:
+            regularization_matrix = np.zeros(
+                (self.data_matrix.shape[1], self.data_matrix.shape[1]),
+            )
+
+        self.regularization_matrix = regularization_matrix
+        self.G_data_weights = self.integration_weights
+        self.G_weights = self.integration_weights
+
+    @_initialize_data.register
+    def _initialize_data_basis(
+        self,
+        data: FDataBasis,
+        integration_weights: None,
+        regularization: Optional[L2Regularization[Any]],
+        weights_basis: Optional[Basis],
+    ) -> None:
+        """Initialize the data of a basis block."""
+        self.mean = data.mean()
+        self.data = data - self.mean
+        self.data_matrix = self.data.coefficients
+
+        # By default, use the basis of the input data
+        # for the weights
+        if weights_basis is None:
+            self.weights_basis = data.basis
+        else:
+            self.weights_basis = weights_basis
+
+        # Compute the regularization matrix
+        # By default, all zeros (no regularization)
+        regularization_matrix = None
+        if regularization is not None:
+            regularization_matrix = compute_penalty_matrix(
+                basis_iterable=(self.weights_basis,),
+                regularization_parameter=1,
+                regularization=regularization,
+            )
+        if regularization_matrix is None:
+            regularization_matrix = np.zeros(
+                (self.data_matrix.shape[1], self.data_matrix.shape[1]),
+            )
+
+        self.regularization_matrix = regularization_matrix
+        self.G_weights = self.weights_basis.gram_matrix()
+        self.G_data_weights = self.data.basis.inner_product_matrix(
+            self.weights_basis,
+        )
+
+    def set_nipals_results(
+        self,
+        rotations: NDArrayFloat,
+        loadings: NDArrayFloat,
+    ) -> None:
+        """Set the results of NIPALS."""
+        self.rotations_matrix = rotations
+        self.loadings_matrix = loadings
+        self.rotations = self._to_block_type(self.rotations_matrix, "rotation")
+        self.loadings = self._to_block_type(self.loadings_matrix, "loading")
+
+    def _to_block_type(
+        self,
+        nipals_matrix: NDArrayFloat,
+        title: str,
+    ) -> BlockType:
+        # Each column of the matrix generated by NIPALS corresponds to
+        # an obsevation or direction. Therefore, they must be transposed
+        # so that each row corresponds ot an observation or direction
+        if isinstance(self.data, FDataGrid):
+            return self.data.copy(
+                data_matrix=nipals_matrix.T,
+                sample_names=[
```

Comment:
I do not really like to modify the labels of the dataset (sample names, coordinate names and dataset name). It is usually unexpected by the user and may cause very ugly names to arise.

Reply:
This is a new dataset. It is created as a copy for convenience, but it is not the original dataset. That is why I changed the dataset name. What would you prefer the labels to be?

Reply:
Then maybe we should create a new one, and not copy.